devito/propagator.py

```diff
@@ -83,11 +85,14 @@ def __init__(self, name, nt, shape, spc_border=0, time_order=0,
 
         # Cache blocking and default block sizes
         self.cache_blocking = cache_blocking
-        if(isinstance(block_size, Iterable)):
-            if(len(block_size) == len(shape)):
+        self.cache_block_inner = cb_inner_dim
+        if isinstance(block_size, Iterable):
+            if len(block_size) == len(shape):
                 self.block_sizes = block_size
+            elif len(block_size) == len(shape) - 1 and not self.cache_block_inner:
```

Comment:
Ok, so there are a few things here I don't quite understand:
a) I am mainly concerned with the complexity of the interface to Propagator and Operator, and I see nothing in your argument that convinces me that an outward facing `cb_inner_dim` parameter is required. Why does the user need to see this exactly?
b) I also don't quite see the need for an explicit flag. Instead could you not insert `None` as the default value into `block_sizes` and check `if block_size is not None` in the loop you quote? This would also allow us more flexibility as to which dimensions to leave open, not just the inner one.
c) Just to understand this better, when auto-tuning you are injecting a separate `block_sizes` array at some point into the Propagator toolchain for each iteration in the parameter space, are you not? Otherwise, how are you re-using the existing cache blocking code?